### PR TITLE
add scanvi

### PR DESCRIPTION
**Pull request by danielStrobl**
## Describe your changes
Added scanVI to batch integration task

## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Bug fixes

- [ ] Proposed changes are described in the CHANGELOG.md

- [x] CI Tests succeed and look good!

### CODE REVIEW

**Comment by rcannood** [comment]
Please let me know when this PR is ready for review ^^

**Comment by danielStrobl** [comment]
@rcannood ready for review

File: src/tasks/batch_integration/methods/scanvi/config.vsh.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    variants:
+      scanvi_full_unscaled:
+      scanvi_hvg_unscaled:
+        hvg: true
+  resources:
+    - type: python_script
+      path: script.py
+platforms:
+  - type: docker
+    image: ghcr.io/openproblems-bio/base_r:1.0.0
```
Comment (KaiWaldrant) [inline src/tasks/batch_integration/methods/scanvi/config.vsh.yaml:37]:
```suggestion
    image: ghcr.io/openproblems-bio/base_r:1.0.1
```

File: src/tasks/batch_integration/methods/scanvi/config.vsh.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    setup:
+      - type: python
+        pypi:
+          - scvi-tools
+          - scanpy
```
Comment (KaiWaldrant) [inline src/tasks/batch_integration/methods/scanvi/config.vsh.yaml:42]:
```suggestion
```
scanpy is now included in base image

File: src/tasks/batch_integration/methods/scanvi/script.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+meta = {
+    'functionality_name' : 'foo',
+    'config': 'bar'
+}
+## VIASH END
+
+with open(meta['config'], 'r', encoding="utf8") as file:
+    config = yaml.safe_load(file)
+
+output_type = config["functionality"]["info"]["subtype"]
```
Comment (KaiWaldrant) [inline src/tasks/batch_integration/methods/scanvi/script.py:20]:
```suggestion
```
no longer required

File: src/tasks/batch_integration/methods/scanvi/script.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    adata = adata[:, adata.var['hvg']].copy()
+
+print('Run scanvi', flush=True)
+adata.X = adata.layers['normalized']
+adata = scanvi(adata, batch='batch', labels='label')
+del adata.X
+
+print("Store outputs", flush=True)
+adata.uns['output_type'] = output_type
+adata.uns['hvg'] = par['hvg']
```
Comment (KaiWaldrant) [inline src/tasks/batch_integration/methods/scanvi/script.py:36]:
```suggestion
```

please remove, will be extracted from config file

File: src/tasks/batch_integration/methods/scanvi/config.vsh.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+functionality:
+  # A unique identifier for your component (required).
+  # Can contain only lowercase letters or underscores.
+  name: scanvi
+
+  # Metadata for your component
+  info:
+    # A relatively short label, used when rendering visualisarions (required)
+    label: Scanvi
```
Comment (KaiWaldrant) [inline src/tasks/batch_integration/methods/scanvi/config.vsh.yaml:16]:
```suggestion
    label: ScanVI
```

File: src/tasks/batch_integration/methods/scanvi/config.vsh.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    # A one sentence summary of how this method works (required). Used when 
+    # rendering summary tables.
+    summary: "ScanVI is an extension of scVI but instead using a Bayesian semi-supervised approach for more principled cell annotation. Run as implemented by Luecken et al. 2022."
+    description : "TODO"
```
Comment (KaiWaldrant) [inline src/tasks/batch_integration/methods/scanvi/config.vsh.yaml:20]: please add description

File: src/tasks/batch_integration/methods/scanvi/script.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    'hvg': True,
+}
+meta = {
+    'functionality_name' : 'foo',
+    'config': 'bar'
+}
+## VIASH END
+
+with open(meta['config'], 'r', encoding="utf8") as file:
+    config = yaml.safe_load(file)
```
Comment (KaiWaldrant) [inline src/tasks/batch_integration/methods/scanvi/script.py:18]:
```suggestion
```

no longer needed to read config due to removal of `output_type`

File: src/tasks/batch_integration/methods/scanvi/config.vsh.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+platforms:
+  - type: docker
+    image: ghcr.io/openproblems-bio/base_r:1.0.0
+    setup:
+      - type: python
+        pypi:
+          - scvi-tools
+          - scanpy
+          - scib==1.1.3
+  - type: nextflow
```
Comment (KaiWaldrant) [inline src/tasks/batch_integration/methods/scanvi/config.vsh.yaml:44]: Add directives for nextflow: you can change the labels to e.g. `highmem` , `midmem` depending on the requirements of the method and how much memory or cpu it needs.

```suggestion
  - type: nextflow
    directives:
      label: [ lowmem, lowcpu ]
```

Comment (KaiWaldrant) [reply src/tasks/batch_integration/methods/scanvi/config.vsh.yaml:44]: It was not in the docs but will be added and in the schemas

**Review by KaiWaldrant — CHANGES_REQUESTED** [review]

**Comment by KaiWaldrant** [comment]
Can you also update the changelog by adding the method to `### V1 MIGRATION` in the task section

**Review by KaiWaldrant — APPROVED** [review]

**Pull request merged**